test/topology_custom/test_mv_topology_change.py

```diff
@@ -193,3 +194,56 @@ async def test_mv_write_to_dead_node(manager: ManagerClient):
+# In this test we perform a write with a view update while the replication factor
+# of the keyspace is being changed and the corresponding base tablet finishes
+# the migration at a different time the view tablet does.
+@pytest.mark.asyncio
+@pytest.mark.parametrize("delayed_replica", ["base", "view"])
+@pytest.mark.parametrize("altered_dc", ["dc1", "dc2"])
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_mv_rf_change(manager: ManagerClient, delayed_replica: str, altered_dc: str):
+    # The workaround for this issue depends on the fact that only last replica
+    # in the DCs replica list is modified when changing RF. Use 2 DCs to ensure
+    # that the fix works even if the new replica is not the last one in the total list.
+    servers = await manager.servers_add(2, property_file={'dc': f'dc1', 'rack': 'myrack'})
+    servers.extend(await manager.servers_add(2, property_file={'dc': f'dc2', 'rack': 'myrack'}))
+
+    cql = manager.get_cql()
+    await cql.run_async("CREATE KEYSPACE IF NOT EXISTS ks WITH replication = {'class': 'NetworkTopologyStrategy', 'dc1': 1, 'dc2': 1} AND tablets = {'initial': 1}")
+    await cql.run_async("CREATE TABLE ks.tab (pk int, ck int,  PRIMARY KEY (pk, ck))")
+    await cql.run_async("CREATE MATERIALIZED VIEW ks.tabv AS SELECT pk, ck FROM ks.tab WHERE ck IS NOT NULL PRIMARY KEY (ck, pk)")
+
+    # Insert a row so that there's data to be streamed during tablet migration
+    await cql.run_async("INSERT INTO ks.tab (pk,ck) VALUES (0,0)")
+
+    await asyncio.gather(*[manager.api.enable_injection(s.ip_addr, f"block_tablet_streaming", True, parameters={'delayed_replica': delayed_replica}) for s in servers])
+
+    ks_fut = cql.run_async(f"ALTER KEYSPACE ks WITH replication = {{'class': 'NetworkTopologyStrategy', '{altered_dc}':2}}")
+    # Perform a write after the migration has started but before it has finished
+    async def first_migration_done():
+        stages = await cql.run_async(f"SELECT stage FROM system.tablets")
+        logger.info(f"Current stages: {[row.stage for row in stages]}")
+        return set([None, "streaming"]) == set([row.stage for row in stages]) or None
+    await wait_for(first_migration_done, time.time() + 60)
+
+    # If the base was delayed, there may be just 1 base replica even though RF is now 2, so use cl=ONE
+    await cql.run_async(SimpleStatement("INSERT INTO ks.tab (pk,ck) VALUES (1,1)", consistency_level=ConsistencyLevel.ONE))
+
+    await asyncio.gather(*[manager.api.message_injection(s.ip_addr, f"block_tablet_streaming") for s in servers])
+    await ks_fut
+
+    # Confirm that the problematic scenario occurred
+    metrics = await asyncio.gather(*[manager.metrics.query(s.ip_addr) for s in servers])
+    replica_mismatches = sum([server_metrics.get('scylla_database_total_base_view_replicas_mismatch') or 0 for server_metrics in metrics])
+    assert replica_mismatches > 0
```

Comment:
Nitpick:
I think a better assertion is that this counter **grew** from its previous value, not that it grew above 0. 
I guess it's the same thing if the cluster is never reused, and this "mismatch" thing can't happen during boot, etc., but why assume that?

Reply:
Ok, I changed it to check if the value increased compared to before the `INSERT` statement